src/Uno.UI/UI/Xaml/VisualStateGroup.cs

```diff
@@ -174,23 +173,11 @@ private void OnParentChanged(object instance, object key, DependencyObjectParent
 
 		private void OnOwnerElementChanged()
 		{
-			if (this.GetParent() is IFrameworkElement { IsParsing: true })
-			{
-				_pendingOnOwnerElementChanged = true;
-				return;
-			}
-
```

Comment:
This feels like something else should break. What about the related tests for these specific lines?

Reply:
@ramezgerges right, so this will revert to a performance issue, which is a regression.